src/mame/drivers/vsmjtria.cpp

```diff
@@ -10,13 +10,22 @@ Unmarked board with everything doubled up, to provide a versus experience with a
 CPU:    2x Z80
 RAM:    6x M5M5117P
 Sound:  2x AY-3-8910
-I/O:    2x 8255, 4x 8-dip banks, 1x single dip
+I/O:    2x 8255, 4x 8-dip banks, 1x single dip switch (at position 11m)
 OSC:    20MHz
 
+Notes:
+- Loosely based off rmhaihai.cpp.
+  Changes needed for merging both implementations seems too trivial to warrant a
+  device merge (basically just the video HW and a few bits).
+
 TODO:
-- coins don't work (set to free play to test for now)
-- VS. feature communications aren't implemented
-- the hardware is based on the one in rmhaihai.cpp. Derive, merge or neither?
+- CPU seems a bit too slow when deciding about what tile to discard (may be btanb);
+- Upper byte of I/O access seems to be some extra comms state machine
+  (i.e. it more or less mimic the commands that CPUs send between each other).
+  Maybe just a debug port that is read thru external HW?
+- Pinpoint how HW reads SW2 (position 11m) single bit dip;
+- Pinpoint how HW reads the three lead-filled marked as SW1 (position 1b);
```

Comment:
SW2 isn’t a DIP switch – it isn’t in a dual-inline package.  It’s just a PCB-mounted slider switch.  It might not actually be read by the CPU at all.  Given how close it is to the battery and non-volatile RAM, it could be for clearing NVRAM.  SW1 looks like a location for another slider switch that isn’t populated for whatever reason.  It’s located right next to the key matrix driver chip, but it’s not clear where the traces go.  Maybe system11 could trace it out?

Reply:
Okay, updated this section over feedback, conversation can be marked as resolved unless there's something additional to add?